### PR TITLE
Adds NOTICE file into rpm files

### PR DESCRIPTION
**Pull request by gurkanindibay**
NOTICE file is added into usr folder in packages
Build fails since citus release branch does not include NOTICE file
you can test it with nightlies.
python -m  tools.packaging_automation.citus_package --gh_token <gh_token> --platform "el/8" --build_type "nightly" --secret_key "${PACKAGING_SECRET_KEY}" --passp
hrase "${PACKAGING_PASSPHRASE}" --output_dir "$(pwd)/packages/" --input_files_dir "$(pwd)" --output_validation
Still working on debian. 
image

- [x] This PR should be merged after addition of NOTICE file into citus project 
citusdata/citus#5479
citusdata/citus#5478
citusdata/citus#5477
citusdata/citus#5476

### CODE REVIEW

File: citus.spec [hunk 1]
```diff
@@ -43,6 +43,7 @@ make %{?_smp_mflags}
 # Install documentation with a better name:
 %{__mkdir} -p %{buildroot}%{pginstdir}/doc/extension
 %{__cp} README.md %{buildroot}%{pginstdir}/doc/extension/README-%{sname}.md
+%{__cp} NOTICE %{buildroot}/usr/NOTICE
```
Comment (hanefi) [inline citus.spec]: Several issues with this:

- the folder `%{buildroot}/usr` may not exist
- we would not want to place a file with a generic name in a directory that is not owned by citus

```suggestion
%{__cp} NOTICE %{buildroot}%{pginstdir}/doc/extension/NOTICE-%{sname}
```

File: pkgvars [hunk 2]
```diff
@@ -2,3 +2,4 @@ pkgname=citus
 pkgdesc='Citus (Open-Source)'
 pkglatest=9.5.10.citus-1
 versioning=fancy
+nightlyref=notice_test
```
Comment (hanefi) [inline pkgvars]: Do we plan to keep this line here? Is it for testing purposes?

BTW, missing EOL character at the end.

File: citus.spec [hunk 3]
```diff
@@ -71,12 +72,14 @@ echo %{pginstdir}/share/extension/%{sname}.control >> installation_files.list
...
 %defattr(-,root,root,-)
 %doc CHANGELOG.md
+
 %if 0%{?rhel} && 0%{?rhel} <= 6
 %doc LICENSE
 %else
 %license LICENSE
 %endif
 %doc %{pginstdir}/doc/extension/README-%{sname}.md
+%doc /usr/NOTICE
```
Comment (hanefi) [inline citus.spec]: I do not like this generic file name that can cause clashes with many other packages.

**Review by hanefi — CHANGES_REQUESTED** [review]

**Comment by hanefi** [comment]
> Build fails since citus release branch does not include NOTICE file


Do you want to place the file on the release branches? You can open a PR on citus and citus enterprise to add the file to default branches, and we can cherry-pick those changes to release branches after the PR is merged.

File: pkgvars (hunk 2 above)
Comment (gurkanindibay) [reply pkgvars]: It is for testing purpose since the master branch does not include NOTICE file. I'm opening a PR to add Notice file into master branch of citus

Comment (hanefi) [reply pkgvars]: We have the file on master. See https://github.com/citusdata/citus/blob/master/NOTICE

Comment (gurkanindibay) [reply pkgvars]: Could we add 9.x release branches? I got error because there is not a NOTICE file for release branches?

File: citus.spec (hunk 1 above)
Comment (gurkanindibay) [reply citus.spec]: Fixed
image

File: citus.spec (hunk 3 above)
Comment (gurkanindibay) [reply citus.spec]: Fixed as you suggested

File: pkgvars (hunk 2 above)
Comment (gurkanindibay) [reply pkgvars]: Branch PR's
https://github.com/citusdata/citus/pull/5479
https://github.com/citusdata/citus/pull/5478
https://github.com/citusdata/citus/pull/5477
https://github.com/citusdata/citus/pull/5476

**Comment by gurkanindibay** [comment]
@onurctirtir  you want me to add a check and ignore Notice file if it doesn't exist. But I think it is not a good practice since adding NOTICE file is Microsoft policy. We need to add this file into our repositories. Anyhow I tried 
I tried adding conditional document addition but conditonal addition is not posswible in prmspec format since conditioning is limited with below cases
image
You can get detailed information from the source below
https://rpm-packaging-guide.github.io/#rpm-conditionals

File: .github/workflows/build-package.yml [hunk 4]
```diff
@@ -81,6 +81,6 @@ jobs:
           version=${pkglatest%.citus-1}
           echo ::set-output name=version::${version}
       - name: Trigger package tests
-        run: gh workflow run "citus-package-tests" --repo citusdata/tools -f prj_ver="${{ steps.set_version.outputs.version }}"
+        run: gh workflow run "citus-package-tests" --repo citusdata/tools -f project_version="${{ steps.set_version.outputs.version }}"
```
Comment (hanefi) [inline .github/workflows/build-package.yml]: split this into multiple lines please.

File: citus.spec [hunk 5]
```diff
@@ -43,6 +43,8 @@ make %{?_smp_mflags}
 # Install documentation with a better name:
 %{__mkdir} -p %{buildroot}%{pginstdir}/doc/extension
 %{__cp} README.md %{buildroot}%{pginstdir}/doc/extension/README-%{sname}.md
+%{__cp} NOTICE %{buildroot}%{pginstdir}/doc/extension/NOTICE-%{sname}
+#%{__cp} NOTICE %{buildroot}/usr/NOTICE
```
Comment (hanefi) [inline citus.spec]: Can you remove commented out lines before sending it for a review?

File: pkgvars [hunk 6]
```diff
@@ -1,4 +1,4 @@
 pkgname=citus
 pkgdesc='Citus (Open-Source)'
 pkglatest=10.0.6.citus-1
-versioning=fancy
```
Comment (hanefi) [inline pkgvars]: Why did you remove the empty line in this file?

**Review by hanefi — CHANGES_REQUESTED** [review]

File: .github/workflows/build-package.yml (hunk 4 above)
Comment (hanefi) [reply .github/workflows/build-package.yml]:
```suggestion
        run: gh workflow run "citus-package-tests" \
             --repo citusdata/tools \
             -f prj_ver="${{ steps.set_version.outputs.version }}"```

File: pkgvars (hunk 6 above)
Comment (gurkanindibay) [reply pkgvars]: Added again

File: citus.spec (hunk 5 above)
Comment (gurkanindibay) [reply citus.spec]: Removed

File: .github/workflows/build-package.yml (hunk 4 above)
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed as you suggested

**Review by hanefi — APPROVED** [review]
